steampipecloud/resource_steampipecloud_workspace_pipeline.go

```diff
@@ -0,0 +1,391 @@
+	}
+	args, err := JSONStringToInterface(d.Get("args").(string))
+	if err != nil {
+		return diag.Errorf("error parsing args for workspace pipeline : %v", d.Get("args").(string))
+	}
+	tags, err := JSONStringToInterface(d.Get("tags").(string))
+	if err != nil {
+		return diag.Errorf("error parsing tags for workspace pipeline : %v", d.Get("tags").(string))
+	}
+	log.Printf("\n[DEBUG] Pipeline Frequency: %v", frequency)
+	log.Printf("\n[DEBUG] Pipeline Arguments: %v", args)
+	log.Printf("\n[DEBUG] Pipeline Tags: %v", tags)
+
+	// Create request
+	req := steampipe.CreatePipelineRequest{Title: title, Pipeline: pipeline, Frequency: frequency, Args: args, Tags: tags}
+
+	userHandle := ""
+	isUser, orgHandle := isUserConnection(d)
+	if isUser {
+		userHandle, r, err = getUserHandler(ctx, client)
+		if err != nil {
+			return diag.Errorf("resourceWorkspacePipelineCreate.getUserHandler error  %v", decodeResponse(r))
+		}
+		resp, r, err = client.APIClient.UserWorkspacePipelines.Create(ctx, userHandle, workspaceHandle).Request(req).Execute()
+	} else {
+		resp, r, err = client.APIClient.OrgWorkspacePipelines.Create(ctx, orgHandle, workspaceHandle).Request(req).Execute()
+	}
+
+	// Error check
+	if err != nil {
+		return diag.Errorf("error creating workspace pipeline: %v", decodeResponse(r))
+	}
+	log.Printf("\n[DEBUG] Pipeline: %s created for Workspace: %s", resp.Id, workspaceHandle)
+
+	// Set property values
+	d.Set("workspace_pipeline_id", resp.Id)
+	d.Set("workspace_id", resp.WorkspaceId)
+	d.Set("title", resp.Title)
+	d.Set("frequency", FormatJson(resp.Frequency))
+	d.Set("pipeline", resp.Pipeline)
+	d.Set("args", FormatJson(resp.Args))
+	d.Set("tags", FormatJson(resp.Tags))
+	d.Set("last_process_id", resp.LastProcessId)
+	d.Set("created_at", resp.CreatedAt)
+	d.Set("updated_at", resp.UpdatedAt)
+	if resp.CreatedBy != nil {
+		d.Set("created_by", resp.CreatedBy.Handle)
+	}
+	if resp.UpdatedBy != nil {
+		d.Set("updated_by", resp.UpdatedBy.Handle)
+	}
+	d.Set("version_id", resp.VersionId)
+	d.Set("organization", orgHandle)
+	d.Set("workspace", workspaceHandle)
+
+	// If Pipeline is created for a Workspace inside an Organization the id will be of the
+	// format "OrganizationHandle/WorkspaceHandle/PipelineID" otherwise "WorkspaceHandle/PipelineID"
+	if userHandle == "" {
+		d.SetId(fmt.Sprintf("%s/%s/%s", orgHandle, workspaceHandle, resp.Id))
+	} else {
+		d.SetId(fmt.Sprintf("%s/%s", workspaceHandle, resp.Id))
+	}
+
+	return diags
+}
+
+func resourceWorkspacePipelineRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*SteampipeClient)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+	var orgHandle, workspaceHandle, pipelineId string
+	var isUser = false
+
+	// If Pipeline is created for a Workspace within an Organization the id will be of the
+	// format "OrganizationHandle/WorkspaceHandle/PipelineID" otherwise "WorkspaceHandle/PipelineID"
```

Comment:
```suggestion
	// If a pipeline is created for a workspace inside an organization then the ID will be of the
	// format "OrganizationHandle/WorkspaceHandle/PipelineID" otherwise "WorkspaceHandle/PipelineID".
```